### ISSUE TITLE
yum repo not showing failed on a package not found

### ISSUE THREAD

**Issue by jjshoe** [ansible/ansible-modules-core#4214]
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

yum module
##### ANSIBLE VERSION

```
ansible 2.1.0.0
  config file = 
  configured module search path = Default w/o overrides

```
##### CONFIGURATION

standard, nothing off the wall
##### OS / ENVIRONMENT

CentOS 7
##### SUMMARY

Output shows `ok` but the task actually failed, very confusing
##### STEPS TO REPRODUCE

```
- name: install some repos
  become: yes
  yum:
    name: "{{ item }}"
    state: latest
  with_items:
    - epel-release
    - docker-io

- name: Install pip, python-dev package and docker with yum
  become: yes
  yum:
    pkg: "{{ item }}"
    state: latest
  with_items:
    - python-dev
    - python-pip
    - docker

- name: Install pip modules
  become: yes
  pip:
    name: "{{ item.name }}"
    version: "{{ item.version | default(omit) }}"
  with_items:
    - name: docker-py
      version: 1.8.1
    - name: docker-compose

- name: enable docker
  become: yes
  service:
    name: docker
    state: started
    enabled: yes

```
##### EXPECTED RESULTS

I would expect the task to fail, python-dev doesn't exist in centos 7
##### ACTUAL RESULTS

```
ok: [xxx.xxx.xxx.xxx] => (item=[u'python-dev', u'python-pip', u'docker']) => {"changed": false, "failed": true, "invocation": {"module_args": {"conf_file": null, "disable_gpg_check": false, "disablerepo": null, "enablerepo": null, "exclude": null, "install_repoquery": true, "list": null, "name": ["python-dev", "python-pip", "docker"], "state": "latest", "update_cache": false, "validate_certs": true}, "module_name": "yum"}, "item": ["python-dev", "python-pip", "docker"], "msg": "No Package matching 'python-dev' found available, installed or updated", "rc": 0, "results": []}
```

**Comment by willthames** [comment]
Minimal test case:

```
- hosts: localhost
  connection: local
  become: yes

  tasks:
  - name: install stuff
    yum:
      name: "{{ item }}"
      state: latest
    with_items:
    - does-not-exist
```

Note that this fails as expected when `with_items` is not used. 

I don't think this is a fault in the yum module necessarily - it's failing correctly - but in the handling of result of the yum module in combination with with_items. However, I suggest keeping this open here until that's sufficiently confirmed that we can move this to ansible/ansible.

**Comment by dagwieers** [comment]
This is not fixed IMO, I implemented PR #4617 to ensure this works for yum.
